docs/content/development/extensions-core/druid-lookups.md

```diff
@@ -0,0 +1,122 @@
+
+### Data Fetcher layer:
+
+First part is the data fetcher layer API `DataFetcher`, that exposes a set of fetch methods to fetch data from the actual Lookup dimension source.
+For instance `JdbcDataFetcher` provides an implementation of `DataFetcher` that can be used to fetch key/value from a RDBMS via JDBC driver.
+If you need new type of data fetcher, all you need to do, is to implement the interface `DataFetcher` and load it via another druid module.
+### Caching layer
+
+This extension comes with two different caching strategies. First strategy is a poll based and the second is a load based.
+#### Poll lookup cache:
+
+The poll strategy cache strategy will fetch and swap all the pair of key/values periodically from the lookup source.
+Hence, user should make sure that the cache can fit all the data. 
+The current implementation provides 2 type of poll cache, the first is onheap (uses immutable map), while the second uses MapBD based offheap map.
+User can also implement a different lookup polling cache by implementing `PollingCacheFactory` and `PollingCache` interfaces. 
+
+#### Loading lookup:
+Loading cache strategy will load the key\value pair upon request on the key it self, the general algorithm is load key if absent.
+Once the key/value  pair is loaded eviction will occur according to the cache eviction policy.
+This module comes with two loading lookup implementation, the first is onheap and based on Guava cache implementation, the second is MapDB offheap implementation.
+Both implementations offer various eviction strategies.
+Same for Loading cache, user can implement a new type of loading cache by implementing `LookupLoadingCache` interface.
+ 
+## Configuration and Operation:
+
+
+### Polling Lookup
+
+**Note that the current implementation of `offHeapPolling` and `onHeapPolling` will create two caches one to lookup value based on key and the other to reverse lookup the key form value**
+
+|Field|Type|Description|Required|default|
+|-----|----|-----------|--------|-------|
+|dataFetcher|Json object|Specifies the lookup data fetcher type  to use in order to fetch data|yes|null|
+|cacheFactory|Json Object|Cache factory implementation|no |onHeapPolling|
+|pollPeriod|Period|polling period |no |null (poll once)|
+
+
+#####   Example of Polling On-heap Lookup
+Case user want to use a polling cache that will swap the content every 10 minutes and using on-heap cache:
+```json
+{
+    "type":"pollingLookup",
+   "pollPeriod":"PT10M",
+   "dataFetcher":{ "type":"jdbcDataFetcher", "connectorConfig":"jdbc://mysql://localhost:3306/my_data_base", "table":"lookup_table_name", "keyColumn":"key_column_name", "valueColumn": "value_column_name"},
+   "cacheFactory":{"type":"onHeapPolling"}
+}
+
+```
+
+#####   Example Polling Off-heap Lookup
+Case polling cache that will fetch the data once and never swap the content since `pollPeriod == null`.
```

Comment:
Suggest something along the lines of:

`This example demonstrates an off-heap lookup that will be cached once and never swapped (pollPeriod == null) `